scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1291,28 +1359,35 @@ class KubeletApi( object ):
         A class for querying the kubelet API
     """
 
-    def __init__( self, k8s, port=10255 ):
+    def __init__( self, k8s, port=10255, host_ip=None ):
         """
         @param k8s - a KubernetesApi object
         """
-        pod_name = k8s.get_pod_name()
-        pod = k8s.query_pod( k8s.namespace, pod_name )
-        spec = pod.get( 'spec', {} )
-        status = pod.get( 'status', {} )
-
-        host_ip = status.get( 'hostIP', None )
-
         if host_ip is None:
-            raise KubeletApiException( "Unable to get host IP for pod: %s/%s" % (k8s.namespace, pod_name) )
+            try:
+                pod_name = k8s.get_pod_name()
+                pod = k8s.query_pod( k8s.namespace, pod_name )
+                status = pod.get( 'status', {} )
+                host_ip = status.get( 'hostIP', None )
+                # Don't raise exception for now
```

Comment:
What was the reason not to raise `KubletApiException` if host IP is null?  Does that reason still hold when merging this back into master?

Reply:
I think this might have been a hangover from a previous iteration of the code.  I did some quick testing, and didn't seem to see any issues with it.  I think we can leave it in.